### PR TITLE
test: hopefully deflaking echo integration test

### LINKED ISSUE #3997: IpVersions/EchoIntegrationTest.AddRemoveListener/IPv6 is flaky
*Description*:
on master 028387a3b0746deaf011ea50104692dfbb8b8d2f run:
`bazel test --runs_per_test=100 //test/integration:echo_integration_test`

will result 2 runs out of 100 `TIMEOUT`, with `"-l trace"` got 8 out of 100.

### PR DESCRIPTION
**Pull request by alyssawilk**
At least one failure mode is that when the listener was released, some other test would yoink the released port, and the "make sure we can not connect to a removed listener" check would unexpectedly result in a connection.  Running the test as exclusive should fix that particular failure mode, and allow us to see if others exist.

I believe the reason the test was flaking more often when run in parallel with -l trace is because the test ran more slowly, the lag between the listener releasing the port and the raw connection driver increased, so the likelihood that another test would snag the port also increased.

*Risk Level*: Low (test only)
*Testing*: 1000 runs with "exclusive"
*Docs Changes*: n/a
*Release Notes*: n/a

Fixes #3997

### CODE REVIEW

**Comment by alyssawilk** [comment]
Yes!!  I could not for the life of me find it - thanks for coming through on bazel as always :-)

It's hard to tell if that works 100% because of course now I can't do my blazing fast 10,000 runs on an overloaded machine, but it really should help on the flakiness front so we'll give it a shot.

**Comment by alyssawilk** [comment]
I've fixed at least one other "test interfering with another test" race so it wouldn't shock me.
They're a pain to debug unless they also interfere with themselves, as this one did.

**Comment by alyssawilk** [comment]
Huh, 2 runs failed.   I'll see if reinstating some of the rawconnection improvements helps.

**Comment by alyssawilk** [comment]
Yeah, allowing for the close-race plus exclusive appears to work better.

File: test/integration/echo_integration_test.cc [hunk 1]
```diff
@@ -118,10 +118,25 @@ TEST_P(EchoIntegrationTest, AddRemoveListener) {
...
+    RawConnectionDriver connection2(
+        new_listener_port, buffer,
+        [&](Network::ClientConnection&, const Buffer::Instance&) -> void { FAIL(); }, version_);
+    connection2.run(Event::Dispatcher::RunType::NonBlock);
+    if (connection2.connection().state() == Network::Connection::State::Closed) {
+      connect_fail = true;
+      break;
+    }
+  }
+  ASSERT_TRUE(connect_fail);
```
Comment (htuch) [inline test/integration/echo_integration_test.cc]: I guess? Would be nice to factor this pattern out if we end up needing it in more than one place :)

**Pull request merged**